kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/PageToken.java

```diff
@@ -0,0 +1,164 @@
+package io.delta.kernel.internal.replay;
+
+import static java.util.Objects.requireNonNull;
+
+import io.delta.kernel.data.Row;
+import io.delta.kernel.internal.data.GenericRow;
+import io.delta.kernel.types.*;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/** Page Token Class for Pagination Support */
+public class PageToken {
+
+  public static PageToken fromRow(Row row) {
+    requireNonNull(row);
+    if (!PAGE_TOKEN_SCHEMA.equals(row.getSchema())) {
+      throw new IllegalArgumentException(
+          "Invalid Page Token: input row schema does not match expected PageToken schema. "
+              + "Expected: "
+              + PAGE_TOKEN_SCHEMA
+              + ", Got: "
+              + row.getSchema());
```

Comment:
I think all of this could fit on one line.
You could also use `String.format` here, too.
Did you want the different schemas printed on different lines? Are you missing some `\n` here?